### DISCUSSION TITLE
Simpler GDPR friendly implementation for Next.js + Tinybird

### DISCUSSION THREAD

**Discussion by grempe** [tinybirdco/web-analytics-starter-kit#35] [Show and tell]
Following up on my suggestions in this feature request:

https://github.com/tinybirdco/web-analytics-starter-kit/issues/32

## Summary

I have created an alternative implementation of the GDPR first-party tracker script meant for use in Next.js apps.

I'd like you to consider incorporating this into this repo and suggesting this as the primary way to host such a setup on Vercel/Next.js as I think its both easier to understand and easier to setup than the current middleware code. It is also safer, and saves unnecessary request to Tinybird.

Benefits
* No separate middleware needs to be published
* No sub-domain needs to exist, or DNS changes made. This event proxy is just a `pages/api/tracking` path on your application.
* Does not expose your Tinybird tracker authentication token to the public.
* More robust and safer code, implemented in Typescript
* Does runtime checks for required fields in the event body, skipping call to Tinybird if not the right shape.
* Will not allow passthrough of additional fields (junk)

I wanted to see if there is any feedback about the implementation approach taken in the following repository.

https://github.com/grempe/nextjs-tinybird-example

I think what should ultimately happen is that this should be incorporated into this repo as the suggested approach for anyone using Next.js apps who want to use the privacy and GDPR friendly tracking enabled by this starter kit.

The current middleware approach still serves a useful function for those who do not use Next.js, but who want a first-party cookie proxy to Tinybird.

There are two main files to look at in the repo:

* https://github.com/grempe/nextjs-tinybird-example/blob/main/pages/api/tracking.ts
* https://github.com/grempe/nextjs-tinybird-example/blob/main/pages/_document.tsx

Cheers.

**Comment by alejandromav** [comment]
This is great, thanks a lot @grempe! 🙌

We’re considering adding an `/examples` folder to the main repo with example for popular frameworks, such as Next.js.
